src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -194,6 +194,17 @@ private LanguageExpression ConvertPropertyAccess(PropertyAccessSyntax propertyAc
                         return this
                             .CreateConverterForIndexReplacement(GetResourceNameSyntax(resourceSymbol), indexExpression, propertyAccess)
                             .ConvertExpression(GetResourceNameSyntax(resourceSymbol));
+                    case "location":
+                        if (resourceSymbol.DeclaringResource.IsExistingResource())
+                        {
+                            // We cannot substitute the value of the existing resource's location because it requires runtime evaluation.
```

Comment:
Our error message text suggests that `location` is an accessible property. Do we need a special error message that tells the user accessing it isn't allowed for an "existing" resource?

Reply:
Yeah makes sense, but I would prefer to do it in another PR. Also we should do it for other top level properties like `kind`, `sku`, etc., although they are less likely to be used.